client/src/App.js

```diff
@@ -0,0 +1,48 @@
+import "./App.css";
+import io from "socket.io-client";
+import { useState } from "react";
+import Chat from "./Chat";
+
+const socket = io.connect("http://localhost:3001");
+
+function App() {
+  const [username, setUsername] = useState("");
+  const [room, setRoom] = useState("");
+  const [showChat, setShowChat] = useState(false);
+
+  const joinRoom = () => {
+    console.log(username, room);
+    if (username !== "" && room !== "") {
+      socket.emit("join_room", room);
+      setShowChat(true);
+    }
+  };
+  return (
+    <div className="App">
+      {!showChat ? (
+        <div className="login">
+          <h3>Join a chat</h3>
+          <input
+            type="text"
+            placeholder="User"
+            onChange={(event) => {
+              setUsername(event.target.value);
+            }}
```

Comment:
curly braces not needed here - onChange={(event) => setUsername(event.target.value)}

Reply:
Resolved